lib/kernel/src/application_controller.erl

```diff
@@ -1799,48 +1805,99 @@ do_config_diff([{Env, Value} | AppEnvNow], AppEnvBefore, {Changed, New}) ->
     end.
 
 
+conf_param_to_conf({config, FileName}) ->
+    BFName = filename:basename(FileName,".config"),
+    FName = filename:join(filename:dirname(FileName),
+                          BFName ++ ".config"),
+    case load_file(FName) of
+        {ok, NewEnv} ->
+            %% OTP-4867 sys.config may now contain names of other
+            %% .config files as well as configuration parameters.
+            %% Therefore read and merge contents.
+            if
+                BFName =:= "sys" ->
+                    DName = filename:dirname(FName),
+                    {ok, SysEnv, Errors} =
+                        check_conf_sys(NewEnv, [], [], DName),
+                    %% Report first error, if any, and terminate
+                    %% (backwards compatible behaviour)
+                    case Errors of
+                        [] ->
+                            SysEnv;
+                        [{error, {SysFName, Line, Str}}|_] ->
+                            throw({error, {SysFName, Line, Str}})
+                    end;
+                true ->
+                    NewEnv
+            end;
+        {error, {Line, _Mod, Str}} ->
+            throw({error, {FName, Line, Str}})
+    end;
+conf_param_to_conf({configfd, FileDescStrP}) ->
+    FileDescStr = unicode:characters_to_nfc_list(FileDescStrP),
+    IsDigit = fun(C) -> lists:member(C, "0123456789") end,
+    FileDesc =
+        try list_to_integer(lists:takewhile(IsDigit, FileDescStr))
```

Comment:
Since you're doing `dropwhile/2` later on, why not `partition/2` instead?

Reply:
That's a good suggestion. I was not familiar with `partition/2` but I have now looked it up and I think it fits very well here.

Reply:
Err, sorry, `splitwith/2`

Reply:
You are right. `partion/2` does not do the right thing but `splitwith/2` does.